cardano-api/src/Cardano/Api/Query.hs

```diff
@@ -105,18 +115,17 @@ data QueryInShelleyBasedEra era result where
        :: QueryInShelleyBasedEra era
             (Map (Hash GenesisKey) ProtocolParametersUpdate)
 
---TODO: add support for these
---     QueryStakeDistribution
---       :: QueryInShelleyBasedEra StakeDistribution
+     QueryStakeDistribution
+       :: QueryInShelleyBasedEra era (Map (Hash StakePoolKey) Rational)
 
      QueryUTxO
        :: Maybe (Set AddressAny)
        -> QueryInShelleyBasedEra era (UTxO era)
 
---     QueryStakeAddresses
---       :: Set StakeAddress
---       -> QueryInShelleyBasedEra (Map StakeAddress Lovelace,
---                                  Map StakeAddress PoolId)
+     QueryStakeAddresses
```

Comment:
@dcoutts  We are duplicating work again: https://github.com/input-output-hk/cardano-node/commit/535504320840194f1a0701ed39c2bc6e6bd575b2

Reply:
Oops. You can take over this PR.